### PR TITLE
docs: Extend LRP guide with troubleshooting section

### PR DESCRIPTION
**Pull request by aditighag**

### CODE REVIEW

**Review by bimmlerd — APPROVED** [review]
CODEOWNERS lgtm

File: Documentation/network/kubernetes/local-redirect-policy.rst [hunk 1]
```diff
@@ -408,6 +408,39 @@ application pods are now redirected to the ``node-local-dns`` pod.
 In the absence of a node-local DNS cache, DNS queries from application pods
 will get directed to cluster DNS pods backed by the ``kube-dns`` service.
 
+* Troubleshooting
```
Comment (ghost) [inline Documentation/network/kubernetes/local-redirect-policy.rst:411]:
```suggestion
Troubleshooting
~~~~~~~~~~~~~~~
```
I'd expect the troubleshooting section to be a nested section, not just a bullet point. But given the deployment instructions are also bullet points, I won't block on it.

File: Documentation/network/kubernetes/local-redirect-policy.rst [hunk 2]
```diff
@@ -408,6 +408,39 @@ application pods are now redirected to the ``node-local-dns`` pod.
...
+
+            $ kubectl exec -it cilium-mhnhz -n kube-system -- cilium-dbg lrp list
+            LRP namespace   LRP name       FrontendType                Matching Service
+            kube-system     nodelocaldns   clusterIP + all svc ports   kube-system/kube-dns
+                            |              10.96.0.10:53/UDP -> 10.244.1.49:53(kube-system/node-local-dns-72r7m),
+                            |              10.96.0.10:53/TCP -> 10.244.1.49:53(kube-system/node-local-dns-72r7m),
+
+        - Check if the corresponding local redirect service entry has been created. If the service entry is missing,
+          there might have been a race condition in applying the policy and the node-local DNS DaemonSet pod resources.
+          As a workaround, you can restart the node-local DNS DaemonSet pods. File a `GitHub issue <https://github.com/cilium/cilium/issues>`_
```
Comment (ghost) [inline Documentation/network/kubernetes/local-redirect-policy.rst]:
```suggestion
          As a workaround, you can restart the node-local DNS DaemonSet pods. File a `GitHub issue <https://github.com/cilium/cilium/issues/new/choose>`_
```
small suggestion

**Review by ghost — APPROVED** [review]
I left some suggestions, but won't block on them.

**Review by ysksuzuki — APPROVED** [review]

File: Documentation/network/kubernetes/local-redirect-policy.rst (hunk 1 above)
Comment (aditighag) [reply Documentation/network/kubernetes/local-redirect-policy.rst:411]: Tried this, the troubleshooting section sort of merges in the block of text above. 

>But given the deployment instructions are also bullet points, I won't block on it.

Right! It's aligned to the other bullet points.

File: Documentation/network/kubernetes/local-redirect-policy.rst (hunk 2 above)
Comment (aditighag) [reply Documentation/network/kubernetes/local-redirect-policy.rst]: Cool... applied!

**Comment by aditighag** [comment]
Documentation only changes, marking it as ready for merge.

**Pull request merged**